### PR TITLE
[Templates] Add article listings for Resources and Support

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This PR implements the article listings for the upcoming Resources and Support (formerly, Learning Center) section of the website. These pages are not Drupal nodes, but rather pages generated during the build. The build outline looks like -

1. All of the Resources and Support articles are pulled into Metalsmith via GraphQL. I had to add some more GraphQL properties onto certain fragments for this work.
1. A dedicated build step for Resources and Support iterates through the Metalsmith files and grabs all of the articles under Resources and Support by checking the `entityBundle`
1. The build step does some grouping and pagination based on category/topics (tags)
1. The build step generates some new files that are added into memory

[Requirements](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/content/tier-2-content-IA-and-design/learning-center-mvp/view-all-articles-in-X-category-landing-pg-requirements.md#all-articles-in-x-lc-category-landing-page)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/14841
https://github.com/department-of-veterans-affairs/va.gov-team/issues/14406

## Testing done
Only tested by observing the generated page in my browser, but I have some ideas for unit tests for us to pursue soon.

### No articles found
This will be the case until launch.

image

### Output
image

### Article listing by category
image

### Article listing by tag
image

## Acceptance criteria
- [ ] Article listings are now generated 

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/stages/build/drupal/graphql/resourcesAndSupportArticleListings.graphql.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        "checklist",
+        "media_list_images",
+        "media_list_videos",
+        "support_resources_detail_page"
+      ]
+    },
+    {
+      field: "status",
+      value: ["1"],
+      enabled: false # $onlyPublishedContent
```
Comment (ncksllvn) [inline src/site/stages/build/drupal/graphql/resourcesAndSupportArticleListings.graphql.js:39]: Testing purposes only - need to remove before merge.

**Comment by kelsonic** [comment]
image

wow... thanks lol 😂

File: src/site/stages/build/drupal/graphql/faqMultipleQa.graphql.js [hunk 2]
```diff
@@ -67,5 +67,20 @@ fragment faqMultipleQA on NodeFaqMultipleQA {
       ... linkTeaser
     }
   }
+  fieldPrimaryCategory {
+    entity {
+      ... on taxonomyTermLcCategories
```
Comment (kelsonic) [inline src/site/stages/build/drupal/graphql/faqMultipleQa.graphql.js:72]: Did you mean to do `on` here? 🙂

File: src/site/stages/build/drupal/graphql/nodeChecklist.graphql.js [hunk 3]
```diff
@@ -53,6 +53,21 @@ fragment nodeChecklist on NodeChecklist {
       }
     }
   }
+  fieldPrimaryCategory {
+    entity {
+      ... on taxonomyTermLcCategories
```
Comment (kelsonic) [inline src/site/stages/build/drupal/graphql/nodeChecklist.graphql.js:58]: Same here

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉 Nice work!!

**Review by jhonny218 — APPROVED** [review]

**Pull request merged**